sklearn/tests/test_base.py

```diff
@@ -436,6 +436,7 @@ def __init__(self, attribute_pickled=5):
 
     def __getstate__(self):
```

Comment:
Yes, that is the intention of the test case. We had written our own serialization logic for estimators, and scikit-learn 0.18 broke all of our modelling code. For what its worth since the BaseEstimator does not contain parameters (and does not contain estimated parameters) scikit-learn shouldn't expect that users inheriting from `BaseEstimator` and overwriting `__getstate__` make that `super(..).__getstate__` call.